submissions/AnnaStoyano/Friends-app/index.js

```diff
@@ -0,0 +1,194 @@
+                ageBtn.classList.add('LowToHigh');
+            } else {
+                app.dipslayCards = app.dipslayCards.sort((a, b) => a.age > b.age ? 1 : -1);
+                ageBtn.classList.remove('LowToHigh');
+            }
+            app.render();
+        });
+    }
+
+    filterByGender() {
+        const app = this;
+        const manCheckbox = document.querySelector('.checkbox-input.man');
+        const womanCheckbox = document.querySelector('.checkbox-input.woman');
+        const genderWrapper = document.querySelector('.menu_item-gender');
+        genderWrapper.addEventListener('click', function () {
+            if (womanCheckbox.checked && manCheckbox.checked) {
+                app.dipslayCards = app.cards;
+            }
+            if (manCheckbox.checked && !womanCheckbox.checked) {
+                app.dipslayCards = app.dipslayCards.filter(card => card.user.gender == 'male');
+            }
+            if (womanCheckbox.checked && !manCheckbox.checked) {
+                app.dipslayCards = app.dipslayCards.filter(card => card.user.gender == 'female');
+            }
+            if (!womanCheckbox.checked && !manCheckbox.checked) {
+                app.dipslayCards = app.cards;
+            }
+            app.render();
+        });
+    }
+
+    filterByCity() {
+        const app = this;
+        const cityInput = document.querySelector('.text-input--city');
+        const cityWrapper = document.querySelector('.menu_item-city');
+        cityWrapper.addEventListener('click', function () {
+            app.prevDisplay = app.dipslayCards;
+        });
+        cityInput.addEventListener('input', function ({target}) {
+            app.dipslayCards = app.dipslayCards.filter(card => card.user['location']['city'].includes(target.value));
+            if (app.dipslayCards.length == 0 || target.value == '') {
+                app.dipslayCards = app.prevDisplay;
+            }
+            app.render();
+        });
+    }
+
+    filterByEmail() {
+        const app = this;
+        const emailInput = document.querySelector('.text-input--email');
+        const emailWrapper = document.querySelector('.menu_item-email');
+        emailWrapper.addEventListener('click', function () {
+            app.prevDisplay = app.dipslayCards;
+        });
+        emailInput.addEventListener('input', function ({target}) {
+            app.dipslayCards = app.dipslayCards.filter(card => card.user.email.includes(target.value));
+            if (app.dipslayCards.length == 0 || target.value == '') {
+                app.dipslayCards = app.prevDisplay;
+            }
+            app.render();
+        });
+    }
+}
+
+class UserCard {
+    constructor(data) {
+        this.card = this.createUserCard();
+        this.name = `${data.name.first} ${data.name.last}`;
+        this.age = `${data['dob'].age} years old`;
+        this.city = data['location'].city.capitalizeFirstLetter();
```

Comment:
I believe CSS rule will be better. Data in this field can be changed later and you can forget to apply this method. Just imaging functionality in a future :)

General rule - everything which can be done with CSS, should be done with CSS.